### ISSUE TITLE
test_https fails with Python 2.7.16

### ISSUE THREAD

**Issue by mcepl** [gevent/gevent#1390]
* gevent version:1.4.0 (building the official openSUSE package)
* Python version: cPython 2.7.16 from openSUSE/Tumbleweed (I am a maintainer of Python packages at SUSE)
* Operating System: openSUSE/Tumbleweed (more details in the attached log)

### Description:

Runnig full testsuite as a part of building openSUSE packages.

```
[  255s]   ======================================================================
[  255s]   ERROR: test_https (__main__.TestUrlopen)
[  255s]   ----------------------------------------------------------------------
[  255s]   Traceback (most recent call last):
[  255s]     File "/tmp/test_urllib2_localnetzJjHGI.py", line 554, in test_https
[  255s]       data = self.urlopen("https://localhost:%s/bizarre" % handler.port, context=context)
[  255s]     File "/tmp/test_urllib2_localnetzJjHGI.py", line 450, in urlopen
[  255s]       f = urllib2.urlopen(url, data, **kwargs)
[  255s]     File "/usr/lib64/python2.7/urllib2.py", line 154, in urlopen
[  255s]       return opener.open(url, data, timeout)
[  255s]     File "/usr/lib64/python2.7/urllib2.py", line 429, in open
[  255s]       response = self._open(req, data)
[  255s]     File "/usr/lib64/python2.7/urllib2.py", line 447, in _open
[  255s]       '_open', req)
[  255s]     File "/usr/lib64/python2.7/urllib2.py", line 407, in _call_chain
[  255s]       result = func(*args)
[  255s]     File "/usr/lib64/python2.7/urllib2.py", line 1241, in https_open
[  255s]       context=self._context)
[  255s]     File "/usr/lib64/python2.7/urllib2.py", line 1198, in do_open
[  255s]       raise URLError(err)
[  255s]   URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)>
```
[Full build log](https://github.com/gevent/gevent/files/3068849/log.txt)

### What I've run:

```sh
cat <<'EOF' >> network_tests.txt
test_urllib2net.py
test__server.py
test__server_pywsgi.py
test__socket_ssl.py
test___example_servers.py
test__socket_dns.py
test__getaddrinfo_import.py
test__examples.py
test_ssl.py
test__ssl.py
EOF
export GEVENT_RESOLVER=thread
# Setting the TRAVIS environment variable makes some different configuration
# for tests that use the network so they don't fail on travis (or obs)
export TRAVIS=1
# Setting the APPVEYOR environment variable makes the tests use a workaround
# for Appveyor that we also need in obs for "wait_threads() failed to cleanup 1 threads"
export APPVEYOR=1
export LANG=en_US.UTF-8
PYTHONPATH=%{buildroot}%{python2_sitearch} python2 -m gevent.tests --ignore network_tests.txt
```

**Comment by jamadden** [comment]
Those tests and certificates come from upstream CPython. I believe they were recently changed to address this, and that will be included in the next gevent release.

**Comment by mcepl** [comment]
Thank you, let me know when this happens, so I can remove the patch from our package.

**Comment by jamadden** [comment]
FWIW, I'm targeting current master (with these fixes) to a gevent 1.5 release, specifically because of the updated cpython versions tested/supported.
